### PR TITLE
Provide more details on how ebpf_bitmap_t works.

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Provide more details on how ebpf_bitmap_t works.

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

File: libs/platform/ebpf_bitmap.c [hunk 1]
```diff
@@ -1,25 +1,54 @@
...
+// 1) Test the bit at position X of N.
+// 2) Set or reset the bit at position X of N, with both interlocked and non-interlocked versions.
+// 3) Iterate over the bits that are set in the bitmap, either in the forward or reverse direction, returning the
+// position of the bit that is set.
+//
+// Iteration over bits:
+// The BitScanForward/BitScanReverse instructions find the least significant bit or most significant bit that is set in
+// a mask (a block of bits) and returns the offset of that bit. To permit iterating over a set bits, this module
+// provides a cursor, which maintains the last offset where a bit was found as well as a copy of the current mask
+// (block) being searched.  At the start of the search, the first block of 64-bits are copied from the bitmap into the
```
Comment (shankarseal) [inline libs/platform/ebpf_bitmap.c:14]: BITS_IN_BLOCK instead of 64.

**Review by shankarseal — APPROVED** [review]

File: libs/platform/ebpf_bitmap.c [hunk 2]
```diff
@@ -1,25 +1,54 @@
...
 
+// This module exposes ebpf_bitmap_t, which maintains a set of N bits. It provides the following operations:
+// 1) Test the bit at position X of N.
+// 2) Set or reset the bit at position X of N, with both interlocked and non-interlocked versions.
+// 3) Iterate over the bits that are set in the bitmap, either in the forward or reverse direction, returning the
+// position of the bit that is set.
+//
+// Iteration over bits:
+// The BitScanForward/BitScanReverse instructions find the least significant bit or most significant bit that is set in
+// a mask (a block of bits) and returns the offset of that bit. To permit iterating over a set bits, this module
```
Comment (dthaler) [inline libs/platform/ebpf_bitmap.c:12]:
```suggestion
// a mask (a block of bits) and return the offset of that bit. To permit iterating over a set of bits, this module
```

File: libs/platform/ebpf_bitmap.c (hunk 1 above)
Comment (dthaler) [inline libs/platform/ebpf_bitmap.c:14]:
```suggestion
// (block) being searched.  At the start of the search, the first block of 64-bits is copied from the bitmap into the
```

File: libs/platform/ebpf_bitmap.c [hunk 3]
```diff
@@ -1,25 +1,54 @@
...
+// provides a cursor, which maintains the last offset where a bit was found as well as a copy of the current mask
+// (block) being searched.  At the start of the search, the first block of 64-bits are copied from the bitmap into the
+// cursor's current block. As each bit in the current block is found and returned, that bit is cleared so that the
+// BitScanForward/BitScanReverse instruction will find the next bit in the current block. Once all the bits in the
+// current block are found, the next block of 64-bits are copied from the bitmap int the cursors current block and the
```
Comment (dthaler) [inline libs/platform/ebpf_bitmap.c:17]:
```suggestion
// current block are found, the next block of 64-bits is copied from the bitmap into the cursor's current block and the
```
("block" is singular)

File: libs/platform/ebpf_bitmap.c [hunk 4]
```diff
@@ -1,25 +1,54 @@
...
+// search is resumed on the new block. Once the cursor's index reaches either 0 (reverse) or bitmaps' bit_count
```
Comment (dthaler) [inline libs/platform/ebpf_bitmap.c:18]:
```suggestion
// search is resumed on the new block. Once the cursor's index reaches either 0 (reverse) or bitmap's bit_count
```

**Review by dthaler — APPROVED** [review]

**Pull request merged**